test/box-luatest/ghs_135_check_iterator_when_loading_gc_consumers_test.lua

```diff
@@ -0,0 +1,39 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_check_iterator_when_loading_gc_consumers = function(cg)
+    t.tarantool.skip_if_not_debug()
+    local server_log_path = g.server:exec(function()
+        return rawget(_G, 'box_cfg_log_file') or box.cfg.log
+    end)
+
+    -- Inject error after recovery - right before WAL GC consumers are loaded.
+    local run_before_cfg = [[
+        local trigger = require('trigger')
+        trigger.set('box.ctl.on_recovery_state', 'errinj', function(state)
+            if state == 'wal_recovered' then
+                box.error.injection.set('ERRINJ_INDEX_ITERATOR_NEW', true)
+            end
+        end)
+    ]]
+    cg.server:restart({
+        env = {
+            ['TARANTOOL_RUN_BEFORE_BOX_CFG'] = run_before_cfg,
+        }
+    }, {wait_until_ready = false})
+    t.helpers.retrying({}, function()
+        t.assert(g.server:grep_log("Failed to recover WAL GC consumers", nil,
+            {filename = server_log_path}))
+    end)
```

Comment:
I guess it is better to check that application panics.

Reply:
Added a check that the application panics. I did it in such way (that's the only way I know):
```lua
t.assert(g.server:grep_log("F> can't initialize storage", nil, {filename = server_log_path}))
```

Reply:
It is not possible using `luatest.server` AFAIK. We usually use `popen` or `justrun` and check exit code.

Reply:
OK - let's leave it as is then. Rewriting the test with `justrun` seems to be overkill in this case.